### ISSUE TITLE
[Usability] The nodeset output cannot tell where the message it coming from

### ISSUE THREAD

**Issue by robin2008** [xcat2/xcat-core#3474]
The message is bad, can't tell where the message it coming from. my assumptions below...
```
[root@fs3 common]# nodeset mid05tor12cn02 offline
mid05tor12cn02: offline
mid05tor12cn02: offline
Error: The mid05tor12cn02 can not be resolved
mid05tor12cn02: offline
Error: The mid05tor12cn02 can not be resolved
```
Note:  need to cover other actions too, for example, stat

Trace https://github.ibm.com/vhu/c910env/issues/77

**Comment by robin2008** [comment]
There are mainly two parts of messages are required be handled.
1, Normal output.
For this kind of output, currently it is using `<node>: <action or stat>`.  without service node, the output could be well handled by `xcoll`, for example,
```
====================================
fake101,fake100
====================================
offline

====================================
fake102,fake103
====================================
install rhels7.3-ppc64le-netboot-compute
```
However, if there are service nodes, the message cannot tell which is from. It is required us to add the information. And there are two options:

1. the format is just like `updatenode`
```
c910f03c05k20: fake100: offline
c910f03c05k20: fake101: offline
c910f03c05k20: fake102: offline
c910f03c05k20: fake103: offline
c910f03c05k20: fake104: offline
c910f03c05k20: fake105: offline
c910f03c05k29: fake104: offline
c910f03c05k29: fake105: offline
```
If using xcoll:
```
====================================
c910f03c05k29
====================================
fake104: offline
fake105: offline

====================================
c910f03c05k20
====================================
fake100: offline
fake101: offline
fake102: offline
fake103: offline
fake104: offline
fake105: offline

```

2. keep the previous format, but append the SN information in each line.
```
fake100: offline (c910f03c05k20)
fake101: offline (c910f03c05k20)
fake102: offline (c910f03c05k20)
fake103: offline (c910f03c05k20)
fake104: offline (c910f03c05k20)
fake105: offline (c910f03c05k20)
fake104: offline (c910f03c05k29)
fake105: offline (c910f03c05k29)
```
If using xcoll:
```
====================================
fake101,fake103,fake100,fake102
====================================
offline (c910f03c05k20)

====================================
fake104,fake105
====================================
offline (c910f03c05k20)
offline (c910f03c05k29)
```
I prefer this option as it looks a bit consistent with previous, also consistent with the case no SNs  ( no SNs, we will not append such information)

BTW:  we can discuss on how to append the SN information there.

2, Error messages
Now error message is generated when handling each node, it's hard to converge together. But I think we could pre-check for known exception case in the begining steps, and then to handle normal process with low risk. In such way, we combine the error message together  to avoid too many error lines.

**Comment by whowutwut** [comment]
If the compute node is managed by a single service node then we should only see one instance of the node in the output? Not 3..? Correct?

**Comment by immarvin** [comment]
hi @whowutwut , according to the current code logic, ``nodeset`` request will be forwarded the MN and all the service nodes in the cluster.

**Comment by robin2008** [comment]
For nodeset/makedhcp,  the current logic is to broadcast to all SN in the cluster. (Not only the servicenode defined for this node)

But we have disjoint mode for this kind of command,  for example, disjointdhcps for makedhcp (already exists), disjointnetboot for nodeset (new introduced in #3088 )

with this settings,  SN only handle their own children, and ignore others.

For for query, we still need a mode to query unmanaged nodes status on this SN in case there are unclean-up case (for example,  remove sn01 for cn01 but forgot to clean the configuration on sn01 )

So I want to add a new argument '-f' to 'stat' sub-command, so that it could query unmanaged CNs on demand.

And we may still require offline to do the same thing to clean-up configurations for unmanaged CNs

**Comment by robin2008** [comment]
Here are some summaries of today's discussion meeting.

1, Try to find a general method for such kinds of output. (to handle it in xcatd), the format should be like:
**`[name:] [(sn):] <info data>`**
**`[name:] [(sn):] <Error data>`**

**Note:**  
   - if the message is coming from myself,  does not output `[(sn):]`
   - the _sn_ could be name or IP if it cannot be resolvable.

2, To enhance nodeset error handling:
  - handle general error CN error before handling normal buz, like DNS lookup.
  - Only show succeed CN in stdout, and report error for all failed nodes together for each SN.

The example might be as below:
```
nodeset fake[1-5] osimage=abc
fake1: install abc
fake2: install abc
fake3: install abc
fake1: (sn01):  install abc
fake2: (sn01): install abc
fake3: (sn02): install abc
Error: Failed to nodeset node(s): fake4,fake5. Check log file for more details.
Error: (sn01) Failed to nodeset node(s): fake3, fake4,fake5. Check log files for more details.
Error: (sn02) Failed to nodeset node(s): fake1, fake3, fake4,fake5. Check log files for more details.
```

**Comment by robin2008** [comment]
As most of xCAT CLI does not have such pain, to reduce the impact to current customer usage. Better to use an on-demand option to enable it.

**New design:**
Highlevel:
1, Every response object will be rendered with the name of running Mn or Sn.
2, Each xCAT Plugin could have the capability to disable it when it does not want to show the name.
3, In CLI side, to show the server name prefix when given option is specified.

Detail:
1, "xcatdsource" is added into each response object by default in xcatd, the value is the name of current server.
2, If user run CLI with environment variable "XCATSHOWSVR",  then the output will be added server name prefix.  The format as below. 
3, If a response with "NoSvrPrefix" key (plugin could set it explicitly), then no extra server name will be shown even if user run it with "XCATSHOWSVR"

```
XCATSHOWSVR=1 nodeset fake1,fake4 osimage=rhels7.3-ppc64le-install-compute
fake1: [c910f03c05k29]: install rhels7.3-ppc64le-compute
fake1: [c910f03c05k20]: install rhels7.3-ppc64le-compute
fake4: [c910f03c05k20]: install rhels7.3-ppc64le-compute
Error: [c910f03c05k20]: The fake4 can not be resolved
```

**Comment by robin2008** [comment]
Fix it by PR #3547.
And it requires test case to cover available CLI,  we need another PR for the test cases.

**Comment by robin2008** [comment]
Test case is covered in PR #3605
